frontend/src/Components/NotificationsPopup.tsx

```diff
@@ -76,50 +79,80 @@ export default function NotificationsPopup(props: NotificationsPopupProps) {
             });
     }, [fetchNotifications, subscribe]);
 
-    const handleNotificationClick = (e: React.MouseEvent<HTMLAnchorElement>, notify: NotificationInfo) => {
+    const handleNotificationClick = (notify: NotificationInfo) => (e: React.MouseEvent<HTMLAnchorElement>) => {
         api.notifications.read(notify.id)
             .then()
             .catch();
+        if (!notify.read) {
+            app.setUnreadNotificationsCount(app.unreadNotificationsCount - 1);
+        }
         if (props.onClose) {
             props.onClose();
         }
     };
 
+    const handleNotificationHide = (notify: NotificationInfo) => (e: React.MouseEvent<unknown>) => {
+        e.preventDefault();
+        e.stopPropagation();
+        api.notifications.hide(notify.id)
+            .then()
+            .catch();
+        if (!notify.read) {
+            app.setUnreadNotificationsCount(app.unreadNotificationsCount - 1);
+        }
+        app.setVisibleNotificationsCount(app.visibleNotificationsCount - 1);
+        setNotifications(notifications?.filter(n => n.id !== notify.id));
+    };
+
     const handleClearAll = () => {
-        api.notifications.readAll()
+        app.setVisibleNotificationsCount(0);
+        app.setUnreadNotificationsCount(0);
+        api.notifications.hideAll()
             .then()
             .catch();
         if (props.onClose) {
             props.onClose();
         }
     };
 
+    const handleReadAll = () => {
+        app.setUnreadNotificationsCount(0);
+        api.notifications.readAll()
+            .then()
+            .catch();
```

Comment:
Output error through `console.error` maybe? Silently swallowing error will make it really hard to debug.